dev-tools/omdb/src/bin/omdb/db/ereport.rs

```diff
@@ -0,0 +1,524 @@
+        .limit(fetch_opts.fetch_limit.get().into())
+        .order_by((sp_dsl::time_collected, sp_dsl::restart_id, sp_dsl::ena))
+        .into_boxed();
+
+    if !args.serials.is_empty() {
+        query =
+            query.filter(sp_dsl::serial_number.eq_any(args.serials.clone()));
+    }
+
+    if !args.ids.is_empty() {
+        query = query.filter(sp_dsl::restart_id.eq_any(args.ids.clone()));
+    }
+    if let Some(before) = args.before {
+        query = query.filter(sp_dsl::time_collected.lt(before));
+    }
+
+    if let Some(after) = args.after {
+        query = query.filter(sp_dsl::time_collected.gt(after));
+    }
+
+    if !fetch_opts.include_deleted {
+        query = query.filter(sp_dsl::time_deleted.is_null());
+    }
+
+    let sp_ereports = query.load_async(&*conn).await.with_context(ctx)?;
+    check_limit(&sp_ereports, fetch_opts.fetch_limit, ctx);
+
+    let ctx = || "loading host OS ereports";
+    let mut query = host_dsl::host_ereport
+        .select(db::model::HostEreport::as_select())
+        .limit(fetch_opts.fetch_limit.get().into())
+        .order_by((
+            host_dsl::time_collected,
+            host_dsl::restart_id,
+            host_dsl::ena,
+        ))
+        .into_boxed();
+
+    if !args.serials.is_empty() {
+        query =
+            query.filter(host_dsl::sled_serial.eq_any(args.serials.clone()));
+    }
+
+    if !args.ids.is_empty() {
+        query = query.filter(host_dsl::restart_id.eq_any(args.ids.clone()));
+    }
+
+    if let Some(before) = args.before {
+        query = query.filter(host_dsl::time_collected.lt(before));
+    }
+
+    if let Some(after) = args.after {
+        query = query.filter(host_dsl::time_collected.gt(after));
+    }
+
+    if !fetch_opts.include_deleted {
+        query = query.filter(host_dsl::time_deleted.is_null());
+    }
+
+    let host_ereports = query.load_async(&*conn).await.with_context(ctx)?;
+    check_limit(&host_ereports, fetch_opts.fetch_limit, ctx);
+
+    let mut rows = sp_ereports
+        .iter()
+        .map(EreportRow::from)
+        .chain(host_ereports.iter().map(EreportRow::from))
+        .collect::<Vec<_>>();
+
+    // Sort everything by time collected so that the host-OS and SP ereports are
+    // interspersed by time collected, reporter, and ENA. Use
+    // `std::cmp::Reverse` so that more recent ereports are displayed first.
+    rows.sort_by_key(|row| {
+        (
+            std::cmp::Reverse(row.time_collected),
+            row.restart_id,
+            std::cmp::Reverse(row.ena),
+        )
+    });
+
+    let mut table = tabled::Table::new(rows);
+    table
+        .with(tabled::settings::Style::empty())
+        .with(tabled::settings::Padding::new(0, 1, 0, 0));
+
+    println!("{table}");
+
+    Ok(())
+}
+
+#[derive(strum::Display)]
+enum SrcType {
+    Switch,
+    Power,
+    #[strum(to_string = "Sled (SP)")]
+    SledSp,
+    // #[strum(to_string = "Sled (OS)")]
+    // SledOS,
```

Comment:
We seem to have "Host OS ereports" in this PR  - does that not qualify as this type of "src"?